OpenRA.Mods.Common/Traits/World/MusicPlaylist.cs

```diff
@@ -54,6 +55,10 @@ public MusicPlaylist(World world, MusicPlaylistInfo info)
 			this.info = info;
 			this.world = world;
 
+			// Reload music with map assets mounted
+			foreach (var entry in world.Map.Rules.Music)
+				entry.Value.Reload();
```

Comment:
This forces the music tracks to be reloaded / reparsed, so I expect this will introduce a jank in the first tick of the game (but I haven't tested this to verify it).  If we absolutely must support this, then it should be done early enough to be hidden by the load screen.